externalNORSERun.py

```diff
@@ -166,72 +154,121 @@
 # Create a matlab structure from the input data given in Matlab doubles
 input_structure = eng.createStructure(f1, 'f', extPBig1, 'extPBig', extXiBig1, 'extXiBig')
 
-# Perform calculation
-eng.PerformCalculation(o, input_structure, nargout=0)
+# Numerical parameters
+# Get the number of rho coordinates
+rho_size = size(coreprof0[0].rho_tor_norm)
 
-# Writing output data to CPOs
+# Initialize arrays for physical parameters
+T_arr = np.zeros(rho_size)
+n_arr = np.zeros(rho_size)
+EHat_arr = np.zeros(rho_size)
+Z_arr = np.zeros(rho_size)
+rhoTor_arr = np.zeros(rho_size)
 
-# Take the data from the NORSE object, wchich will go into the CPO.
-distribution = np.array(eng.extractDistribution(o)).tolist()
-pBig = np.array(eng.extractPBig(o)).tolist()
-xiBig = np.array(eng.extractXiBig(o)).tolist()
+# Constant physical parameters
+for i in range(rho_size):
+	
+	# Fill physics arrays with values from CPOs
+	T_arr[i] = readIn.convert(coreprof0[0].te.value, i)					# eV
+	n_arr[i] = readIn.convert(coreprof0[0].ne.value, i)					# m^{-3}
+	EHat_arr[i] = EHat.calculate(n_arr[i],CoulombLogarithm.calculate(n_arr[i],T_arr [i]),readIn.convert(coreprof0[0].profiles1d.eparallel.value, i))			# E/E_c
+	Z_arr[i] = readIn.convert(coreprof0[0].profiles1d.zeff.value, i)			# Z_eff
+	rhoTor_arr[i] = readIn.convert(coreprof0[0].rho_tor, i)					# m
 
-# flatten the data to python list, so it can be given to the CPO
-distribution = list(itertools.chain.from_iterable(distribution))
-pBig = list(itertools.chain.from_iterable(pBig))
-xiBig = list(itertools.chain.from_iterable(xiBig))
+B = readIn.convert(coreprof0[0].toroid_field.b0)						# T

```

Comment:
It is to be discussed what kind of B is needed here, but for the sake of generality it should also be a vector.